File: Reverse Polish Notation LeetCode 150/Solution.py
from math import ceil, floor
class Solution:
    def evalRPN(self, tokens):
        stack = []
        for t in tokens:
            if t in '+-*/':
                b, a = stack.pop(), stack.pop()
                if t == '+':
                    stack.append(a+b)
                if t == '-':
                    stack.append(a-b)
                if t == '*':
                    stack.append(a*b)
                else:
                    div = a / b
                    if (div < 0):
                        stack.append(ceil(div))
                    else:
                        stack.append(floor(div))
            else:
                stack.append(int(t))
                
        return stack[0]
    
sol = Solution()
tokens = ["2","1","+","3","*"]
value = sol.evalRPN(tokens)
print(value)
                    

File: Reverse Polish Notation LeetCode 150/Solution.java
import java.util.Stack;

public class Solution {
    public static void main(String[] args){
        Solution1 sol = new Solution1();
        String[] tokens = {"2","1","+","3","*"};
        int value = sol.evalRPN(tokens);
        System.out.println(value);

    }
}

class Solution1 {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();

        for (String token : tokens){
            if (token.equals("+") || token.equals("-") || token.equals("*") || token.equals("/")) {
                int b = stack.pop();
                int a = stack.pop();
                switch (token) {
                case "+":
                    stack.push(a + b);
                    break;
                case "-":
                    stack.push(a - b);
                    break;
                case "*":
                    stack.push(a * b);
                    break;
                case "/":
                    double division = (double) a / b;
                    stack.push(division < 0 ? (int) Math.ceil(division) : (int) Math.floor(division));
                    break;
                        } 
            }
            else {
                stack.push(Integer.parseInt(token));
            }
        }


        return stack.pop();
    }
}